### DISCUSSION TITLE
How to refetch all dependent queries

### DISCUSSION THREAD

**Discussion by pbt4991** [TanStack/query#2612] [Q&A] [answered]
I'm not sure if this is even the "React Query" way of doing things, but let's say:
- I have a query A whose key contains a result of query B. 
- Similarly, the key of query B contains a result from query C.

CSB Example: https://codesandbox.io/s/loving-firefly-q2ent?file=/src/App.js

If I want to refetch A to get the latest data, I actually need to refetch C, which triggers a refetch of B, and eventually A.
Currently, there's no way for me to easily do that without having to reach down to query C manually, right?

**Comment by TkDodo** [comment, marked as answer]
have them start with the same query key and then use `queryClient.invalidateQueries`, like:

```
['myData', 'a', b]
['myData', 'b', c]
['myData', 'c']
```

```
and then call `queryClient.invalidateQueries(['myData'])
```
